### ISSUE TITLE
The bookmark button is missing, sometimes

### ISSUE THREAD

**Issue by skronenwerth** [advplyr/audiobookshelf-app#643]
### Steps to reproduce

1. Start to play an audiobook
2. Continuous playback for about one hour, maybe a litte bit longer
3. The bookmark button is then missing from the audiobook

### Expected behaviour
- That I can still bookmark audiobooks after an hour of playing them continuously.

### Actual behaviour
- When I listen to an audiobook for a longer time, the bookmark button is no longer there.
- The effect occurs sometime after one or two hours of continuous listening. The audiobook is streamed, whether it really occurs every time I don't know, but it is very frequent.
- When I start another audiobook, the button for the other audiobook is there again. If I then start the original audiobook again, it is still gone for that book. This can only be remedied by restarting the app.
- This is how it should look:

![IMG_5400](https://user-images.githubusercontent.com/786982/227118829-5614c0ee-9cef-41b7-baeb-e0f07538d342.jpeg)

- This is how it looks when the button is missing: 

![IMG_5399](https://user-images.githubusercontent.com/786982/227118897-e659f5e2-dffc-4fdf-8849-3c15e378adf6.jpeg)

### Environment data

Audiobookshelf Version:

- [ ] Android App?
- [X] iOS App v0.9.63

#### iOS Issue
iOS Version: 16.3.1

iPhone model: 11

**Comment by skronenwerth** [comment]
No, the audiobooks are heard in stream from the server. I do not have downloads, because it does not yet work the way I need it.

**Comment by advplyr** [comment]
The button disappears when there is no connection to the server because offline bookmarks isn't supported yet. I haven't yet tested on iOS but this does work correctly on Android.

To test on Android I started listening with connection and confirmed the bookmark icon is there. Then I turn on airplane mode and the bookmark button is hidden. Turning off airplane mode will show the bookmark icon again.

The next time the bookmark button is gone, you can minimize the audio player and check the connection icon in the app bar at the top. If it is red then the bookmark button will not be there.

**Comment by skronenwerth** [comment]
Today I had the problem again, airplane mode was off all the time. The bookmark button is gone again, I minimized the player and the connection icon is on online:

![IMG_5582](https://github.com/advplyr/audiobookshelf-app/assets/786982/f0734d56-443d-4d7f-9072-9bbdd3807448)

If I close the app completely and then open it again and continue the audiobook the button is still gone. However, if I briefly start another book and then go back to the original book and start it, the button is there again. This behavior is reproducible as soon as the error occurred. Everything I hear by direct play.

![IMG_5583](https://github.com/advplyr/audiobookshelf-app/assets/786982/990dd26a-9c99-4ed6-a0cc-68144d995404)

**Comment by emacsbliss** [comment]
any ETA for the offline bookmark feature for iOS? I feel this is quite important since often times I may not have internet connection (or unstable) like in subway.. 

Thanks a lot for the great work! look forward to this feature being implemented.

**Comment by CyberAustin** [comment]
Just wanted to add this is still a thing. On Android.

**Comment by stevemckenney** [comment]
Can confirm this still on Android as well. I always download my media, but between the Android Auto sync issues, and no bookmark when offline, i'm consistently losing progress in some fashion using the android app
